detox/src/utils/ConfigurationResolver.test.js

```diff
@@ -0,0 +1,57 @@
+const fs = require('fs');
```

Comment:
You know... since the number of possible run environments is limited to 4, it would make sense to create four folders in `__mocks__/configuration-resolver-it/`, e.g.:

* `__mocks__/configuration-resolver-it/detoxrc-and-package/`
* `__mocks__/configuration-resolver-it/detoxrc/`
* `__mocks__/configuration-resolver-it/package/`
* `__mocks__/configuration-resolver-it/blank/`

I think tests would be more readable then, with less hassle about creating the environment dynamically.
Also, there are no tests for unhappy paths when a resolved file is not found or is malformed.